examples/howto/optimize_evoked.py

```diff
@@ -12,203 +12,278 @@
-                       params['evprox2_nmda_L5_pyramidal']}
+    weights_ampa_p2 = {'L2_basket': 0.000003, 'L2_pyramidal': 1.438840,
+                       'L5_basket': 0.008958, 'L5_pyramidal': params['evprox2_ampa_L5_pyramidal']}
     net.add_evoked_drive('evprox2',
                          mu=params['evprox2_mu'],
                          sigma=params['evprox2_sigma'],
                          numspikes=1,
                          location='proximal',
                          weights_ampa=weights_ampa_p2,
-                         weights_nmda=weights_nmda_p2,
-                         synaptic_delays=synaptic_delays_p)
+                         synaptic_delays=synaptic_delays_p,
+                         n_drive_cells=n_drive_cells,
+                         cell_specific=cell_specific)
 
 ###############################################################################
-# Then, we define the constraints.
+# 5. Define optimization constraints
+# ----------------------------------
 #
-# The constraints must be a dictionary of tuples where the first value in each
-# tuple is the lower bound and the second value is the upper bound for the
-# corresponding parameter.
+# The constraints define the permissible range (lower and upper bounds) for each
+# parameter during the optimization process.
 #
-# The following synaptic weight parameter ranges (units of micro-siemens)
-# were chosen so as to keep the model in physiologically realistic regimes.
+# The ranges for synaptic weights (µS) are set to maintain physiologically
+# realistic model behavior, rather than solely relying on existing literature values.
 
 
-constraints = dict({'evprox2_ampa_L2_basket': (0.01, 1.),
-                    'evprox2_ampa_L2_pyramidal': (0.01, 1.),
-                    'evprox2_ampa_L5_basket': (0.01, 1.),
-                    'evprox2_ampa_L5_pyramidal': (0.01, 1.),
-                    'evprox2_nmda_L2_basket': (0.01, 1.),
-                    'evprox2_nmda_L2_pyramidal': (0.01, 1.),
-                    'evprox2_nmda_L5_basket': (0.01, 1.),
-                    'evprox2_nmda_L5_pyramidal': (0.01, 1.),
-                    'evprox2_mu': (100., 120.),
-                    'evprox2_sigma': (2., 30.)})
+lower_bound = 0.1 * 10**-1
+upper_bound = 0.1 * 10
+constraints = dict({'evprox2_ampa_L5_pyramidal': (lower_bound, upper_bound),
+                    'evprox2_mu': (100., 150.),
+                    'evprox2_sigma': (1., 20.)})
 
 ###############################################################################
-# Now we define and fit the optimizer.
+# 6. Define the initial parameters
+# --------------------------------
+#
+# For optimal results, set initial_params to your hand-tuned values since they
+# already provide a good fit.
+#
+# If initial_params is not set, the optimizer will use the midpoint of the
+# constraints as the initial parameters.
 
-from hnn_core.optimization import Optimizer
+initial_params = dict({'evprox2_ampa_L5_pyramidal': 0.1,
```

Comment:
I'm a little confused how this gets used in the `run_opt()` functions. I see that they are passed as arguments to the objective functions, but I don't see any new changes that really use `initial_params`

Was `initial_params` a variable used internally and now this is just exposing the ability to set the value?